src/Listeners/MailMessage/CreateMailExecutedSubscriber.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace FluxErp\Listeners\MailMessage;
+
+use FluxErp\Actions\Comment\CreateComment;
+use FluxErp\Actions\MailMessage\CreateMailMessage;
+use FluxErp\Actions\PurchaseInvoice\CreatePurchaseInvoice;
+use FluxErp\Actions\Ticket\CreateTicket;
+use FluxErp\Models\Address;
+use FluxErp\Models\Client;
+use FluxErp\Models\Communication;
+use FluxErp\Models\ContactOption;
+use FluxErp\Models\Currency;
+use FluxErp\Models\Media;
+use Illuminate\Support\Str;
+use Spatie\Activitylog\Facades\CauserResolver;
+
+class CreateMailExecutedSubscriber
+{
+    protected ?Address $address = null;
+
+    public function handle(CreateMailMessage $event): void
+    {
+        $message = $event->getResult();
+        $this->address = $this->findAddress($message);
+
+        if ($message->mailFolder->creates_purchase_invoice && $message->media()->count() !== 0) {
+            $this->createPurchaseInvoice($message);
+        }
+
+        $matches = [];
+        preg_match(
+            '/\[flux:comment:(\w+):(\d+)]/',
+            $message->text_body ?? $message->html_body ?? $message->subject,
+            $matches
+        );
+        if (count($matches) === 3 && $message->mailFolder->creates_ticket) {
+            $model = $matches[1];
+            $id = $matches[2];
+
+            try {
+                CreateComment::make([
+                    'model_type' => $model,
+                    'model_id' => $id,
+                    'comment' => $message->text_body ?? $message->html_body ?? $message->subject,
+                    'is_internal' => false,
+                ])->validate()->execute();
+            } catch (\Throwable) {
+            }
+        } elseif ($message->mailFolder->creates_ticket) {
+            $this->createTicket($message);
+        }
+    }
+
+    public function subscribe(): array
+    {
+        return [
+            'action.executed: ' . resolve_static(CreateMailMessage::class, 'class') => 'handle',
+        ];
+    }
+
+    protected function createPurchaseInvoice(Communication $message): void
+    {
+        $contact = $this->address->contact;
```

Comment:
$this->address can be null, so this may throw an exception.
$this->address?->contact